crates/sparrow-transforms/src/transforms/project.rs

```diff
@@ -0,0 +1,51 @@
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+use error_stack::{IntoReport, ResultExt};
+use sparrow_arrow::Batch;
+
+use crate::transforms::Transform;
+use crate::Error;
+use sparrow_expressions::ExpressionExecutor;
+
+/// Transform for projection.
```

Comment:
Most things should look more this -- a simple transform with no actual synchronization logic. Basically, we should use "pipelines" when we need to interact with how things are scheduled (merge, repartition, transform) and implement these kind of simpler "transforms" for the bulk of "batch in -> do a thing -> batch out".